### PR TITLE
fix: 4-projects - SA creation

### LINKED ISSUE #15: Seperate Project Creation and Service Account creation
4-projects folder creates similar issue
[Issue on Service Account Module](https://github.com/terraform-google-modules/terraform-google-service-accounts/issues/24)

### PR DESCRIPTION
**Pull request by rutalreja-deloitte**
Fixes #15 
Tested full apply of 4-projects with no errors

Modified-
1. 4-projects//dev/non-prod/prod - SImilar changes
2. 4-projects//shared

### CODE REVIEW

**Comment by rutalreja-deloitte** [comment]
Also removes deprecation warning in previous PR Lint tests
@bharathkkb please review and merge

File: app-foundation/4-projects/development/boa_gke.tf [hunk 1]
```diff
@@ -66,14 +78,22 @@ module "boa_gke_project" {
...
+  project  = module.boa_gke_project.project_id
+  role     = each.value
+  member   = "serviceAccount:${google_service_account.boa_gsa_sa.email}"
+}
+
+resource "google_project_iam_member" "boa_gsa_sa_roles_sql" {
+  for_each = toset(local.gsa_sa_roles.sql)
+  project  = module.boa_sql_project.project_id
+  role     = each.value
+  member   = "serviceAccount:${google_service_account.boa_gsa_sa.email}"
```
Comment (bharathkkb) [inline app-foundation/4-projects/development/boa_gke.tf:98]: can we for_each over `project_id=[roles]`

File: app-foundation/4-projects/development/boa_secret.tf [hunk 2]
```diff
@@ -31,32 +31,31 @@ locals {
       "roles/iam.roleAdmin",
       "roles/binaryauthorization.policyEditor"
     ],
-    "${module.boa_ops_project.project_id}" = [
```
Comment (bharathkkb) [inline app-foundation/4-projects/development/boa_secret.tf:34]: Same comment as above, can't we keep this as it?

File: app-foundation/4-projects/development/boa_gke.tf (hunk 1 above)
Comment (bharathkkb) [reply app-foundation/4-projects/development/boa_gke.tf:98]: resolved offline - keeping as is due to `The "for_each" value depends on resource attributes that cannot be determined
until apply`

File: app-foundation/4-projects/development/boa_secret.tf (hunk 2 above)
Comment (bharathkkb) [reply app-foundation/4-projects/development/boa_secret.tf:34]: resolved offline - keeping as is due to `The "for_each" value depends on resource attributes that cannot be determined
until apply`

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**